lib/devise/migratable/encryptors/base.rb

```diff
@@ -0,0 +1,27 @@
+module Devise
+    # Implements a way of adding different encryptions.
+    # The class should implement a self.digest method that taks the following params:
+    #   - password
+    #   - stretches: the number of times the encryption will be applied
+    #   - salt: the password salt as defined by devise
+    #   - pepper: Devise config option
+    #
+    module Migratable
```

Comment:
I am wondering why we need to replicate the same logic as the `Encryptable` [module](https://github.com/betterup/devise-encryptable/blob/mike/migratable/lib/devise/encryptable/encryptors/base.rb). I understand you are trying to isolate the changes and have a way to gradually migrate. But that can just be hooked into the normal structure as it was before. The more we move away and duplicate functionality from the original gem, the harder it is to follow upstream. From what I understand, all you had to do in order to have the gradual migration functionality is to hook into `encryptable/model.rb` logic. As it is, it seems that we have replicated the gem inside the gem.

Reply:
Great question. That's the first thing I did, trying to squeeze the "migratable thinking" into the existing `Encryptable` module. But there are just too many corner cases needs to be taken care of, which I think eventually developing very complex logic I don't like to maintain.

One example of this is the goal of `Migratable` is to _never_ touch the existing `encrypted_password` column until in the later phase we want to just swap the column values, which I think is a less risk and less messy solution comparing to do bunch of if/else then decide wether to mutate the `encrypted_password` column or not.

Another reason is there is really not much duplicating of logic comparing to current `devise-encrytable`'s `main` branch. I think keep them separate and setting different expectation and purpose making the maintaince and adoption easier in general.

Reply:
Hey @29decibel, just checking to see what you decided with respect to this issue we talked last time.

Reply:
Yeah, I plan to move this out into a separate better up gem once the testing on K8 finished. Let me know what do you think? After that, whether to contribute back to open-source we can decide later.